File: SoftRhPrendasBO/src/pe/edu/pucp/softrh/prendas/bo/PromocionBO.java
package pe.edu.pucp.softrh.prendas.bo;

import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softrh.prendas.dao.PromocionDAO;
import pe.edu.pucp.softrh.prendas.daoimp.PromocionDAOImp;
import pe.edu.pucp.softrh.prendas.model.Prenda;
import pe.edu.pucp.softrh.prendas.model.Promocion;
import pe.edu.pucp.softrh.prendas.model.TipoPromocion;
import pe.edu.pucp.softrh.usuarios.model.Trabajador;

public class PromocionBO {
    private PromocionDAO promocionDAO;

    public PromocionBO(){
        this.promocionDAO = new PromocionDAOImp();
    }

	// String nombre, String descripcion, Double valorDescuento, TipoPromocion tipo, Date fechaInicio, Date fechaFin, Trabajador trabajador, Prenda prenda
    public Integer insertar(String nombre, String descripcion, Double valorDescuento, TipoPromocion tipo, Date fechaInicio, Date fechaFin, Trabajador trabajador, Prenda prenda){
        Promocion promocion = new Promocion(nombre, descripcion, valorDescuento, tipo, fechaInicio, fechaFin, trabajador, prenda);
        return this.promocionDAO.insertar(promocion);
    }

    public Integer modificar(Integer idPromocion, String nombre, String descripcion, Double valorDescuento, TipoPromocion tipo, Date fechaInicio, Date fechaFin, Trabajador trabajador, Prenda prenda){ //la prenda ya modificada
		Promocion promocion = new Promocion(nombre, descripcion, valorDescuento, tipo, fechaInicio, fechaFin, trabajador, prenda);
		promocion.setIdPromocion(idPromocion);
        return this.promocionDAO.modificar(promocion);
    }

    public Integer eliminar(Integer idPromocion){
        return this.promocionDAO.eliminar(idPromocion);
    }

    public ArrayList<Promocion> listarTodos() {
        return this.promocionDAO.listarTodos();
    }

    public Promocion obtenerPorId(Integer idPromocion) {
        return this.promocionDAO.obtenerPorId(idPromocion);
    }
}

File: Front/RHStoreWS/RHStorePrendasBO/PromocionBO.cs
using RHStoreBaseBO;
using RHStoreBaseBO.ServiciosWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RHStorePrendasBO
{
	public class PromocionBO : BaseBO
	{
        public int insertar(string nombre, string descripcion, double valorDescuento, tipoPromocion tipo, DateTime fechaInicio, DateTime fechaFin, trabajador _trabajador, prenda _prenda)
        {
            return PromocionWS.insertarPromocion(nombre, descripcion, valorDescuento, tipo, fechaInicio, fechaFin, _trabajador, _prenda);
        }
        public int modificar(int idPromocion, string nombre, string descripcion, double valorDescuento, tipoPromocion tipo, DateTime fechaInicio, DateTime fechaFin, trabajador _trabajador, prenda _prenda)
        {
            return PromocionWS.modificarPromocion(idPromocion, nombre, descripcion, valorDescuento, tipo, fechaInicio, fechaFin, _trabajador, _prenda);
        }
        public int eliminar(int idPromocion)
        {
            return PromocionWS.eliminarPromocion(idPromocion);
        }

        public BindingList<promocion> listarTodos()
        {
            return new BindingList<promocion>(PromocionWS.listarPromocionesTodas());
        }

        public promocion obtenerPorId(int idPromocion)
        {
            return PromocionWS.obtenerPromocionPorId(idPromocion);
        }

		public BindingList<promocion> listarPorNombre(string nombre)
		{
			return new BindingList<promocion>(PromocionWS.listarPromocionesPorNombre(nombre));
		}
	}
}
